sdks/java/extensions/sql/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -153,4 +190,115 @@ SqlDrop SqlDropTable(Span s, boolean replace) :
     }
 }
 
+Schema.FieldType FieldType() :
+{
+    final SqlTypeName collectionTypeName;
+    Schema.FieldType fieldType;
+    final Span s = Span.of();
+}
+{
+    (
+        fieldType = Map()
+    |
+        fieldType = Array()
+    |
+        fieldType = Row()
+    |
+        fieldType = SimpleType()
+    )
+    [
+        collectionTypeName = CollectionTypeName()
```

Comment:
<!--new_thread; commit:acf50611c0f2b329b63e3a21ccc9e54db51e5d97; resolved:0-->
This is to support `INT ARRAY` in that syntax for it? TBH I don't think we need to start with that, even though it is "standard" all the related dialects don't use that syntax.

And I didn't realize JavaCC generated LL(k) parsers. That's unfortunate. So I can see why the grammar has to be factored like this. But what this doesn't support is (INT ARRAY) ARRAY or the no-parens version INT ARRAY ARRAY.

I suppose it is fine to leave as-is, but probably want to file something about the limitation, or tell me that I am wrong. And lots of tests. And also feel free to remove if you don't want the complication.

Reply:
Yes this clause is for the postfix array support. I left it mostly because it was already kinda there, just limited to `MULTISETS`, not arrays. We don't support either at the moment, and I would rather remove it for now for consistency